android/src/main/java/com/pauldemarco/flutter_blue/FlutterBluePlugin.java

```diff
@@ -679,6 +682,10 @@ private ScanCallback getScanCallback21() {
                 @Override
                 public void onScanResult(int callbackType, ScanResult result) {
                     super.onScanResult(callbackType, result);
+                    if (!allowDuplicate && result != null && result.getDevice() != null && result.getDevice().getAddress() != null) {
```

Comment:
I guess `allowDuplicate` should be `allowDuplicates` here (with the additional `s`)?

```suggestion
                    if (!allowDuplicates && result != null && result.getDevice() != null && result.getDevice().getAddress() != null) {
```